[Debug Mode] Show reason for RBR/GBR on LHN tabs

pac-guerreiro opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$https://github.com/Expensify/App/issues/49600
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

<details>

<summary>To trigger a GBR in Inbox</summary>

**Make sure there are no reports with RBR**
1. Open any report from `Inbox` page
2. Click on the `+` on the chat editor and select `Assign Task`
3. Fill all the necessary data and assign the task to your self
4. The inbox should now show a green indicator and a message saying 'There's a report awaiting action' above the bottom bar
5. Open another report, then click on `View` button after the message 
6. You should be redirected to the report where you assigned the task to yourself

</details>

<details>

<summary>To trigger a RBR in Inbox</summary>

1. Create a new expense
2. Remove a required property from expense (e.g. `Category`)
3. The inbox indicator should be red
4. A message saying 'There's a report with errors' should be shown above the bottom bar
5. If you click on `View`, you should be redirected to the expense report

</details>

<details>

<summary>To trigger a GBR in Settings</summary>

1. Go to `Profile`, then open `Contact method`
2. Add new contact method
3. The new contact method should show a GBR indicator, so as settings avatar in the bottom bar
4. A message saying 'A contact method requires verification' should be displayed above the bottom bar
5. If you click on `View`, you should be redirected to the contact method page

</details>

<details>

<summary>To trigger a RBR in Settings</summary>

1. Go to `Profile`, then open `Contact method`
6. Add new contact method
8. During verification phase enter any code, then continue
9. The new contact method should show a RBR indicator, so as settings avatar in the bottom bar
10. A message saying 'There's a problem with a contact method' should be displayed above the bottom bar
11. If you click on `View`, you should be redirected to the contact method page

</details>

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as tests

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Same as tests

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/8e7b0508-89ce-4fa0-a579-920559a43604

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/595c6cb2-b3ba-4bc5-b3e6-d21009f5182a

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/d36362a9-de8e-459a-a320-7403c9b2a702

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/54c9473c-75c1-4437-a087-d25f4c142761

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/00437c6d-8130-44f6-9496-86ebf5e36e4e

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
https://github.com/user-attachments/assets/afcd5d5d-68e5-4914-b7bf-8bb85bb16a5a

</details>

pac-guerreiro committed:
chore(debug): initial implementation for LHN and settings GBR/RBR debug

pac-guerreiro commented:
I need to add unit tests for the new hook that I created - `useSettingsStatus` - to make sure that it's working property since I extracted it's logic from `Indicator.tsx`, so it could be reused in debug mode.

Besides this, I also have to:

- Finish english translations
- Translate english to spanish
- Make `useSettingsStatus` return and optional `policyID` which is necessary to redirect user's to the policy pages that contain errors or require attention
- Test if the translations fit in the UI
- Take screenshots and add them to the PR
- Add testing steps to the PR

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

# Conflicts:
#	src/languages/en.ts

pac-guerreiro committed:
chore(debug): add missing translations and routes for settings debugging UI

pac-guerreiro commented:
Today's update:

- [x] Finish english translations
- [x] Make useSettingsStatus return and optional policyID which is necessary to redirect user's to the policy pages that contain errors or require attention

To continue tomorrow:
- [ ] Translate english to spanish
- [ ] Add unit tests for `useSettingStatus` hook
- [ ] Test if the translations fit in the UI
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

pac-guerreiro committed:
style: apply linter

fabioh8010 commented on src/libs/WorkspacesSettingsUtils.ts:
```diff
@@ -129,25 +129,16 @@ function getChatTabBrickRoad(policyID?: string): BrickRoad | undefined {
-    if (hasChatTabRBR) {
-        return CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
-    }
-
-    if (hasChatTabGBR) {
-        return CONST.BRICK_ROAD_INDICATOR_STATUS.INFO;
-    }
-
-    return undefined;
+function getChatTabBrickRoad(policyID?: string): BrickRoad | undefined {
+    const report = getChatTabBrickRoadReport(policyID);
+    return report ? getBrickRoadForPolicy(report) : undefined;
```
What happened to the previous logic? Are we maintaining it somewhere ?

pac-guerreiro replied:
@fabioh8010 I got confused with my own code hehehe but I managed to restore the previous logic!

Thanks for this, great catch 😄

fabioh8010 commented on src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:
```diff
@@ -0,0 +1,167 @@
+                    src={Expensicons.DotIndicator}
+                    fill={indicator}
+                />
+                <Text style={[styles.textWhite, styles.lh20]}>{message && translate(message)}</Text>
```
```suggestion
                {message && <Text style={[styles.textWhite, styles.lh20]}>{translate(message)}</Text>}
```

fabioh8010 commented on src/CONST.ts:
```diff
@@ -5776,6 +5776,21 @@ const CONST = {
         TAGS_ARTICLE_LINK: 'https://help.expensify.com/articles/expensify-classic/workspaces/Create-tags#import-a-spreadsheet-1',
     },
 
+    SETTINGS_STATUS: {
```
I think this should be called `INDICATOR_STATUS`

fabioh8010 commented on src/hooks/useSettingsStatus.ts:
```diff
@@ -0,0 +1,74 @@
+import ONYXKEYS from '@src/ONYXKEYS';
+import useTheme from './useTheme';
+
+type SettingsStatus = {
+    indicatorColor: string;
+    status: ValueOf<typeof CONST.SETTINGS_STATUS> | undefined;
+    idOfPolicyWithErrors: string | undefined;
+};
+
+function useSettingsStatus(): SettingsStatus {
```
```suggestion
type IndicatorStatus = {
    indicatorColor: string;
    status: ValueOf<typeof CONST.SETTINGS_STATUS> | undefined;
    idOfPolicyWithErrors: string | undefined;
};

function useIndicatorStatus(): IndicatorStatus {
```

Correct if I'm wrong, but shouldn't this be called `useIndicatorStatus` since its main purpose is to be used in the `Indicator` component?

If yes I think we should also rename `SETTINGS_STATUS` to `INDICATOR_STATUS`

fabioh8010 commented on src/languages/en.ts:
```diff
@@ -4899,6 +4899,22 @@ const translations = {
         date: 'Date',
         time: 'Time',
         none: 'None',
+        settingsStatus: {
```
```suggestion
        indicatorStatus: {
```

Probably should be `indicatorStatus`

fabioh8010 reviewed (commented):
Partial review

pac-guerreiro committed:
chore: add tests for useIndicatorStatus

pac-guerreiro committed:
refactor: apply suggestions

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

# Conflicts:
#	src/CONST.ts
#	src/languages/en.ts
#	src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx

pac-guerreiro commented:
Today's update:
- [x] Add unit tests for `useSettingStatus` hook
- [x] Addressed @fabioh8010's feedback

To continue tomorrow:
- [ ] Translate english to spanish
- [ ] Test if the translations fit in the UI
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

pac-guerreiro committed:
chore: apply suggestions

pac-guerreiro committed:
refactor: simplify test structure of useIndicatorStatusTest

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

pac-guerreiro committed:
fix(debug mode): message not fitting in the UI

pac-guerreiro committed:
fix(debug mode): wrong routes used to redirect user to settings problems

pac-guerreiro committed:
feat(debug mode): improve english translations

pac-guerreiro commented:
Today's update:
- [x] Improved english translations
- [x] Fixed wrong routes being used for contact method error or info
- [x] Fixed GBR/RBR message not fitting in the UI

Todo:
- [ ] Test subscription GBR/RBR, reimbursement account RBR, workspace employee list RBR and workspace custom unit RBR
- [ ] Translate english to spanish
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

fabioh8010 commented on src/hooks/useIndicatorStatus.ts:
```diff
@@ -0,0 +1,74 @@
+    // All of the error & info-checking methods are put into an array. This is so that using _.some() will return
+    // early as soon as the first error / info condition is returned. This makes the checks very efficient since
+    // we only care if a single error / info condition exists anywhere.
+    const errorChecking = {
```
```suggestion
    const errorChecking: Partial<Record<IndicatorStatus, boolean>> = {
```

fabioh8010 commented on src/hooks/useIndicatorStatus.ts:
```diff
@@ -0,0 +1,74 @@
+        [CONST.INDICATOR_STATUS.HAS_WALLET_TERMS_ERRORS]: Object.keys(walletTerms?.errors ?? {}).length > 0 && !walletTerms?.chatReportID,
+    };
+
+    const infoChecking = {
```
```suggestion
    const infoChecking: Partial<Record<IndicatorStatus, boolean>> = {
```

fabioh8010 commented on src/hooks/useIndicatorStatus.ts:
```diff
@@ -0,0 +1,74 @@
+import ONYXKEYS from '@src/ONYXKEYS';
+import useTheme from './useTheme';
+
+type IndicatorStatus = {
+    indicatorColor: string;
+    status: ValueOf<typeof CONST.INDICATOR_STATUS> | undefined;
+    idOfPolicyWithErrors: string | undefined;
+};
+
+function useIndicatorStatus(): IndicatorStatus {
```
```suggestion
type IndicatorStatus = ValueOf<typeof CONST.INDICATOR_STATUS>;

type IndicatorStatusResult = {
    indicatorColor: string;
    status: IndicatorStatus | undefined;
    policyIDWithErrors: string | undefined;
};

function useIndicatorStatus(): IndicatorStatusResult {
```

So you can reuse `IndicatorStatus` below and in other files

fabioh8010 commented on src/hooks/useIndicatorStatus.ts:
```diff
@@ -0,0 +1,74 @@
+    const [error] = Object.entries(errorChecking).find(([, value]) => value) ?? [];
+    const [info] = Object.entries(infoChecking).find(([, value]) => value) ?? [];
+
+    const status = (error ?? info) as ValueOf<typeof CONST.INDICATOR_STATUS> | undefined;
```
```suggestion
    const status = (error ?? info) as IndicatorStatus | undefined;
```

fabioh8010 commented on src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:
```diff
@@ -0,0 +1,167 @@
+    activeWorkspaceID?: string;
+};
+
+function getSettingsMessage(status: ValueOf<typeof CONST.INDICATOR_STATUS> | undefined): TranslationPaths | undefined {
```
```suggestion
function getSettingsMessage(status: IndicatorStatus | undefined): TranslationPaths | undefined {
```

fabioh8010 commented on src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:
```diff
@@ -0,0 +1,167 @@
+    }
+}
+
+function getSettingsRoute(status: ValueOf<typeof CONST.INDICATOR_STATUS> | undefined, idOfPolicyWithErrors = ''): Route | undefined {
```
```suggestion
function getSettingsRoute(status: IndicatorStatus | undefined, policyIDWithErrors = ''): Route | undefined {
```

fabioh8010 commented on src/libs/WorkspacesSettingsUtils.ts:
```diff
@@ -120,7 +120,7 @@ function hasWorkspaceSettingsRBR(policy: Policy) {
 }
 
-function getChatTabBrickRoad(policyID?: string): BrickRoad | undefined {
+function getChatTabBrickRoadReport(policyID?: string) {
```
```suggestion
function getChatTabBrickRoadReport(policyID?: string): OnyxEntry<Report> {
```

fabioh8010 commented on src/libs/WorkspacesSettingsUtils.ts:
```diff
@@ -129,25 +129,16 @@ function getChatTabBrickRoad(policyID?: string): BrickRoad | undefined {
-            hasChatTabGBR = true;
-        }
-        return brickRoad === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+        // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+        return [CONST.BRICK_ROAD_INDICATOR_STATUS.INFO, CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR].includes(brickRoad as NonNullable<BrickRoad>);
```
```suggestion
        if (!brickRoad) {
            return false;
        }
        return [CONST.BRICK_ROAD_INDICATOR_STATUS.INFO, CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR].includes(brickRoad);
```

fabioh8010 commented on tests/unit/useIndicatorStatusTest.tsx:
```diff
@@ -0,0 +1,239 @@
+import ONYXKEYS from '@src/ONYXKEYS';
+import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
+
+const getMockForStatus = (status: ValueOf<typeof CONST.INDICATOR_STATUS>) =>
```
```suggestion
const getMockForStatus = (status: IndicatorStatus) =>
```

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

pac-guerreiro committed:
refactor(debug mode): apply suggestions

pac-guerreiro commented:
I forgot to leave my update yesterday, here it is:
- [x] Test subscription GBR/RBR, reimbursement account RBR, workspace employee list RBR and workspace custom unit RBR
- [x] Improve english translations
- [x] Translate english to spanish
- [x] Apply @fabioh8010's feedback

Todo:
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

pac-guerreiro committed:
chore(debug mode): improve english translations and add spanish translations

pac-guerreiro committed:
fix(debug mode): incorrect route for reimbursement account problems

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

pac-guerreiro committed:
chore: remove old todo

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

pac-guerreiro commented:
All tasks done. This PR is now ready for review!

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

fabioh8010 reviewed (commented):
Minor comments

pac-guerreiro committed:
refactor: apply suggestions

pac-guerreiro commented:
Still waiting on final review from @DylanDylann

pac-guerreiro commented:
Still waiting on final review from @DylanDylann

DylanDylann commented:
https://github.com/user-attachments/assets/d8e85696-c3e4-47ec-a2f4-2e3d9d54cc1d

Let's use text color to be suitable with the background color

DylanDylann commented on src/languages/es.ts:
```diff
@@ -5573,6 +5572,22 @@ const translations = {
             hasChildReportAwaitingAction: 'Informe secundario pendiente de acción',
             hasMissingInvoiceBankAccount: 'Falta la cuenta bancaria de la factura',
         },
+        indicatorStatus: {
```
@pac-guerreiro NAB: Did you confirm these translations?

DylanDylann commented on src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:
```diff
@@ -0,0 +1,172 @@
+function getSettingsRoute(status: IndicatorStatus | undefined, reimbursementAccount: OnyxEntry<ReimbursementAccount>, policyIDWithErrors = ''): Route | undefined {
+    switch (status) {
+        case CONST.INDICATOR_STATUS.HAS_CUSTOM_UNITS_ERROR:
+            return ROUTES.WORKSPACE_INITIAL.getRoute(policyIDWithErrors);
```
```suggestion
            return ROUTES.WORKSPACE_DISTANCE_RATES.getRoute(policyIDWithErrors);
```

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

pac-guerreiro committed:
chore(debug mode): apply suggestions

- Fix text color in debug tab view when using light theme
- Update route for custom units error

pac-guerreiro commented:
@DylanDylann all feedback addressed!

I'll let you know when someone confirms the spanish translations 😄

DylanDylann commented:
@pac-guerreiro We have conflicts, please help to resolve it

DylanDylann commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

DylanDylann commented on src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx:
```diff
@@ -109,67 +112,80 @@ function BottomTabBar({selectedTab}: BottomTabBarProps) {
-                    <Text
-                        style={[styles.textSmall, styles.textAlignCenter, styles.mt1Half, selectedTab === SCREENS.HOME ? styles.textBold : styles.textSupporting, styles.bottomTabBarLabel]}
+        <>
+            <DebugTabView
```
@pac-guerreiro  Only display this if users enable the debug feature

pac-guerreiro committed:
Merge branch 'main' into pac-guerreiro/feature/49600-show-reason-for-rbr-gbr-on-lhn-tabs

# Conflicts:
#	src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx

pac-guerreiro committed:
fix(debug mode): showing debug tab view when debug mode is disabled

pac-guerreiro commented:
@DylanDylann I just resolved the conflict and applied a fix for the problem you encountered 😄

pac-guerreiro commented:
Thank you @DylanDylann for the review! 😄

The PR is ready to be merged!

puneetlath reviewed (approved):
This looks great and very thorough. Thank you!

puneetlath commented:
Hmm, all the tests are passing.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.
